fix: resolve CI workflow formatting issues

Sandijigs opened the pull request:
Fixed contract test formatting to pass scarb fmt --check validation.
closes  #132

Sandijigs committed:
fix: resolve CI workflow formatting issues

Fixed contract test formatting to pass scarb fmt --check validation.

Sandijigs committed:
fixed CI workflow formatting issues

Sandijigs committed:
Merge branch 'main' into fix/workflow-ci-formatting

Sandijigs commented:
@SudiptaPaul-31 i have fixed the issue.

SudiptaPaul-31 commented:
<img width="1064" height="397" alt="image" src="https://github.com/user-attachments/assets/8f3ae958-d85c-43cc-a094-eb67b4b2c997" />
@Sandijigs

Sandijigs committed:
fix: resolve CI workflow and dependency issues

- Pin React to 18.3.1 to fix @starknet-react/core compatibility
- Fix Scarb installation in CI workflow with correct version flag
- Add npm optimization flags for faster CI builds

Sandijigs committed:
fix: regenerate package-lock.json with correct React 18.3.1 dependencies

- Remove old package-lock.json entries causing React 19 conflicts
- Ensure @starknet-react/core@4.0.4 uses React 18.3.1 consistently
- Fix ERESOLVE dependency conflicts in CI
